File: examples/backends/python/scribe/log.py
"""Three log levels on stderr, in the same line format as the other backends.

`logging` would do, but its configuration would be longer than this file, and uvicorn
already owns the root of that tree.
"""

import sys
from datetime import UTC, datetime


def _write(level: str, message: str) -> None:
    now = datetime.now(UTC).isoformat(timespec="milliseconds").replace("+00:00", "Z")
    print(f"{now} {level:>5} scribe_backend: {message}", file=sys.stderr, flush=True)


def info(message: str) -> None:
    _write("INFO", message)


def warn(message: str) -> None:
    _write("WARN", message)


def error(message: str) -> None:
    _write("ERROR", message)

File: examples/backends/typescript/src/log.ts
/**
 * Three log levels on stderr. A logging library would be the second dependency of a
 * backend whose first and only one is the Meteroid SDK.
 */

function write(level: string, message: string): void {
  process.stderr.write(`${new Date().toISOString()} ${level.padStart(5)} scribe_backend: ${message}\n`);
}

export const log = {
  info: (message: string) => write("INFO", message),
  warn: (message: string) => write("WARN", message),
  error: (message: string) => write("ERROR", message),
};
